hazelcast/src/main/java/com/hazelcast/config/MemberAddressProviderConfig.java

```diff
@@ -101,25 +107,20 @@ public boolean equals(Object o) {
 
         MemberAddressProviderConfig that = (MemberAddressProviderConfig) o;
 
-        if (isEnabled() != that.isEnabled()) {
-            return false;
-        }
-        if (getClassName() != null ? !getClassName().equals(that.getClassName()) : that.getClassName() != null) {
-            return false;
-        }
-        if (!getProperties().equals(that.getProperties())) {
-            return false;
+        return isEnabled() == that.isEnabled()
+            && getProperties().equals(that.getProperties())
+            && Objects.equals(implementationNameInternal(), that.implementationNameInternal());
+    }
+
+    private String implementationNameInternal() {
```

Comment:
`implementationNameInternal` is still in use, so implementation of `equals/hashCode` is inconsistent with real object content.